lib/srv/db/common/audit.go

```diff
@@ -80,12 +90,18 @@ func NewAudit(config AuditConfig) (Audit, error) {
 	}
 	return &audit{
 		cfg: config,
-		log: logrus.WithField(trace.Component, "db:audit"),
+		log: logrus.WithField(trace.Component, config.Component),
 	}, nil
 }
 
+func (a *audit) methodCallMetrics() func() {
+	return methodCallMetrics(a.cfg.Component, a.cfg.Engine, 2)
+}
+
 // OnSessionStart emits an audit event when database session starts.
 func (a *audit) OnSessionStart(ctx context.Context, session *Session, sessionErr error) {
+	defer a.methodCallMetrics()()
```

Comment:
the `methodCallMetrics` is also called in bellow in `a.EmitEvent(ctx, event)`
What do we want to measure here ?

Reply:
Yeah, I wondered about it too. I landed on the current design because both methods are callable independently and the actual metrics incremented are different because their method names are different. They will be highly correlated, though.

Reply:
We now only measure `EmitEvent`